src/OVAL/probes/unix/process58.c

```diff
@@ -351,12 +351,110 @@ static int get_exec_shield_status(int pid) {
 	return ret;
 }
 
+struct dynamic_buffer{
+	char* mem;
+	size_t size;
+};
+
+/**
+ * Parse /proc/%d/cmdline file
+ * @param filepath Path to file ~ use preallocated buffer for the path
+ * @param buffer output buffer with non-zero size
+ * @return ps-like command info or NULL
+ */
+static inline bool get_process_cmdline(const char* filepath, struct dynamic_buffer* const  buffer){
+
+	assert(buffer->size > 0);
+
+	int fd = open(filepath, O_RDONLY, 0);
+
+	if (fd < 0) {
+		return false;
+	}
+
+	size_t used = 0;
+	ssize_t remaining_size;
+	ssize_t read_size;
+
+	for(;;) {
+
+		// Read data, store to buffer
+		remaining_size = buffer->size - used - 1; // for trailing zeros
+		read_size = read(fd, buffer->mem + used, remaining_size );
+		used += read_size;
+
+		// If reach end of file, then end the loop
+		if (remaining_size != read_size) {
+			break;
+		}
+
+		buffer->size *= 2;
+		// Double size of buffer
+		char* new_mem = realloc(buffer->mem, sizeof(char) * buffer->size );
+		if ( new_mem == NULL ){
+			close(fd);
+			return false;
+		}
+		buffer->mem = new_mem;
+	}
+
+	close(fd);
+
+	if ( used == 0 ) { // empty file
+		return false;
+	} else {
+
+		buffer->mem[used] = '\0';
+		/**
+		 * Skip multiple trailing zeros
+		 */
+		int i = used - 1;
+		while ( (i > 0) && (buffer->mem[i] == '\0') ) {
+			--i;
+		}
+
+		/**
+		 * Program and args are separated by '\0'
+		 * Replace them with spaces ' '
+		 */
+		while( i >= 0){
+			if ( buffer->mem[i] == '\0' ) {
+				buffer->mem[i] = ' ';
+			}
+			--i;
+		}
+	}	
+	return true;
+}
+
+/**
+ * Make "[%s] <defunct>" from cmd string - inplace
+ * @param cmd_buffer @see read_process() > cmd_buffer
+ * @return pointer to start of string
+ */
+static inline char *make_defunc_str(char* const cmd_buffer){
+	static const char DEFUNC_STR[] = "] <defunct>";
```

Comment:
I think this syntax isn't portable. Is there any reason why we shouldn't just use `static const char*` instead?